### PR TITLE
Run integration tests on ECK

### PR DESCRIPTION
**Pull request by kuisathaverat**
## What does this PR do?

This PR is a new pipeline that deploys an ECK cluster per each test and runs the integration tests in it.

## Why is it important?

It is a follow up of  https://github.com/elastic/apm-integration-testing/pull/700, it will start a daily test of APM on an ECK deploy.

## Related issues
Closes https://github.com/elastic/apm-integration-testing/issues/575 

**NOTE** it should be merged after https://github.com/elastic/apm-integration-testing/pull/700 both PRs have commonly modified files.

### CODE REVIEW

File: .ci/integrationTestECK.groovy [hunk 1]
```diff
@@ -0,0 +1,199 @@
...
+    [var: 'ES_USER', password: es.stringData.user],
+    [var: 'ES_PASS', password: es.stringData.password],
+    [var: 'KIBANA_URL', password: kb.stringData.url],
+    [var: 'BUILD_OPTS', password: "${params.BUILD_OPTS} --apm-server-url ${apm.stringData.url} --apm-server-secret-token ${apm.stringData.token}"]
+  ]){
+    body()
+  }
+}
+
+def wrappingup(label){
```
Comment (mdelapenya) [inline .ci/integrationTestECK.groovy]: Camelcase?

File: .ci/integrationTestEC.groovy [hunk 2]
```diff
@@ -0,0 +1,198 @@
...
+    [var: 'ES_USER', password: es.stringData.user],
+    [var: 'ES_PASS', password: es.stringData.password],
+    [var: 'KIBANA_URL', password: kb.stringData.url],
+    [var: 'BUILD_OPTS', password: "${params.BUILD_OPTS} --apm-server-url ${apm.stringData.url} --apm-server-secret-token ${apm.stringData.token}"]
+  ]){
+    body()
+  }
+}
+
+def wrappingup(label){
```
Comment (mdelapenya) [inline .ci/integrationTestEC.groovy]: Camelcase?

File: Makefile [hunk 3]
```diff
@@ -124,7 +127,9 @@ dockerized-test:
 	  -e RAILS_URL=$(RAILS_URL) \
 	  -e RUM_URL=$(RUM_URL) \
 	  -e PYTHONDONTWRITEBYTECODE=1 \
-	  -e ENABLE_ES_DUMP=$(ENABLE_ES_DUMP) \
+		-e ENABLE_ES_DUMP=$(ENABLE_ES_DUMP) \
```
Comment (mdelapenya) [inline Makefile]: Better use same indentation

File: .ci/integrationTestEC.groovy [hunk 4]
```diff
@@ -0,0 +1,198 @@
...
+        }
+        axes {
+          axis {
+              name 'TEST'
+              values 'all', 'dotnet', 'go', 'java', 'nodejs', 'python', 'ruby', 'rum'
+          }
+          axis {
+              name 'ELASTIC_STACK_VERSION'
+              values '8.0.0-SNAPSHOT', '7.5.1-SNAPSHOT', '7.4.3-SNAPSHOT'
+          }
```
Comment (mdelapenya) [inline .ci/integrationTestEC.groovy]: Is it possible to get these values from a file, to be reused across the two new pipelines?

File: .ci/integrationTestEC.groovy [hunk 5]
```diff
@@ -0,0 +1,198 @@
...
+    EC_DIR="src/github.com/elastic/observability-test-environments"
+    NOTIFY_TO = credentials('notify-to')
+    JOB_GCS_BUCKET = credentials('gcs-bucket')
+    PIPELINE_LOG_LEVEL='DEBUG'
+    DOCKERELASTIC_SECRET = 'secret/apm-team/ci/docker-registry/prod'
+    DOCKER_REGISTRY = 'docker.elastic.co'
+  }
+  triggers {
+    cron 'H H(3-4) * * 1-5'
+    issueCommentTrigger('(?i).*(?:jenkins\\W+)?run\\W+(?:the\\W+)?tests(?:\\W+please)?.*')
```
Comment (v1v) [inline .ci/integrationTestEC.groovy]:
```suggestion
    issueCommentTrigger('(?i).*jenkins\\W+run\\W+(?:the\\W+)?tests(?:\\W+please)?.*')
```

File: .ci/integrationTestEC.groovy [hunk 6]
```diff
@@ -0,0 +1,198 @@
...
+    }
+  }
+  post {
+    cleanup {
+      notifyBuildResult()
+    }
+  }
+}
+
+def withVaultEnv(Closure body){
```
Comment (v1v) [inline .ci/integrationTestEC.groovy]: This is not a blocker but a todo to be moved to the library

File: .ci/integrationTestECK.groovy [hunk 7]
```diff
@@ -0,0 +1,199 @@
...
+    EC_DIR="src/github.com/elastic/observability-test-environments"
+    NOTIFY_TO = credentials('notify-to')
+    JOB_GCS_BUCKET = credentials('gcs-bucket')
+    PIPELINE_LOG_LEVEL='DEBUG'
+    DOCKERELASTIC_SECRET = 'secret/apm-team/ci/docker-registry/prod'
+    DOCKER_REGISTRY = 'docker.elastic.co'
+  }
+  triggers {
+    cron 'H H(3-4) * * 1-5'
+    issueCommentTrigger('(?i).*(?:jenkins\\W+)?run\\W+(?:the\\W+)?tests(?:\\W+please)?.*')
```
Comment (v1v) [inline .ci/integrationTestECK.groovy]:
```suggestion
    issueCommentTrigger('(?i).*jenkins\\W+run\\W+(?:the\\W+)?tests(?:\\W+please)?.*')
```

File: Makefile [hunk 8]
```diff
@@ -112,9 +115,9 @@ dockerized-test:
 	  --name=apm-integration-testing \
 	  --network=apm-integration-testing \
 	  --security-opt seccomp=unconfined \
-	  -e APM_SERVER_URL=$(APM_SERVER_URL) \
-	  -e ES_URL=$(ES_URL) \
-	  -e KIBANA_URL=$(KIBANA_URL) \
+	  -e APM_SERVER_URL=${APM_SERVER_URL} \
```
Comment (v1v) [inline Makefile]:
```suggestion
	  -e APM_SERVER_URL=$${APM_SERVER_URL} \
```

File: .ci/integrationTestEC.groovy (hunk 6 above)
Comment (mdelapenya) [reply .ci/integrationTestEC.groovy]: Not to forget this, we have this issue: https://github.com/elastic/observability-dev/issues/39

File: .ci/integrationTestEC.groovy (hunk 4 above)
Comment (kuisathaverat) [reply .ci/integrationTestEC.groovy]: no, it is not possible for now

File: Makefile (hunk 3 above)
Comment (kuisathaverat) [reply Makefile]: I've fixed it in the other PR

File: .ci/integrationTestECK.groovy [hunk 9]
```diff
@@ -0,0 +1,199 @@
...
+    catchError(buildResult: 'SUCCESS', stageResult: 'SUCCESS') {
+      if(deployConfig.k8s.enabled){
+        sh(label: 'Destroy k8s cluster', script: 'make destroy-cluster')
+      }
+    }
+    catchError(buildResult: 'SUCCESS', stageResult: 'SUCCESS') {
+      if(deployConfig.elasticsearch.type == 'ec'){
+        withVaultToken(){
+          retry(3){
+            sh(label: 'Destroy EC clsuter', script: 'make -C elastic-cloud set-auth-env destroy-cluster')
```
Comment (v1v) [inline .ci/integrationTestECK.groovy]:
```suggestion
            sh(label: 'Destroy EC cluster', script: 'make -C elastic-cloud set-auth-env destroy-cluster')
```

File: .ci/integrationTestEC.groovy [hunk 10]
```diff
@@ -0,0 +1,199 @@
...
+  stages {
+    /**
+     Checkout the code and stash it, to use it on other stages.
+    */
+    stage('Checkout'){
+      steps {
+        deleteDir()
+        gitCheckout(basedir: "${BASE_DIR}")
+        dir("${EC_DIR}"){
+          git(branch: 'master-v2.0',
```
Comment (kuisathaverat) [inline .ci/integrationTestEC.groovy]:
```suggestion
          git(branch: 'master',
```

File: .ci/integrationTestEC.groovy [hunk 11]
```diff
@@ -0,0 +1,199 @@
...
+          ENABLE_ES_DUMP = "true"
+        }
+        axes {
+          axis {
+              name 'TEST'
+              values 'all', 'dotnet', 'go', 'java', 'nodejs', 'python', 'ruby', 'rum'
+          }
+          axis {
+              name 'ELASTIC_STACK_VERSION'
+              values '8.0.0-SNAPSHOT', '7.5.1-SNAPSHOT', '7.4.3-SNAPSHOT'
```
Comment (kuisathaverat) [inline .ci/integrationTestEC.groovy]:
```suggestion
              values '8.0.0-SNAPSHOT', '7.6.0-SNAPSHOT', '7.5.3-SNAPSHOT'
```

File: .ci/integrationTestECK.groovy [hunk 12]
```diff
@@ -0,0 +1,175 @@
...
+  stages {
+    /**
+     Checkout the code and stash it, to use it on other stages.
+    */
+    stage('Checkout'){
+      steps {
+        deleteDir()
+        gitCheckout(basedir: "${BASE_DIR}")
+        dir("${EC_DIR}"){
+          git(branch: 'master-v2.0',
```
Comment (kuisathaverat) [inline .ci/integrationTestECK.groovy]:
```suggestion
          git(branch: 'master',
```

**Pull request merged**